utils/train_utils.py

```diff
@@ -174,7 +177,7 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
         results['avg_eval_prep'] = avg_eval_prep
         results['avg_eval_loss'] = avg_eval_loss
         
-
+    dist.barrier()
```

Comment:
Please check if this is still required, since the barrier has been added before and after the checkpoint already.